### PR TITLE
Make Dask CUDA work with the new WorkerMemoryManager abstraction

### PR DESCRIPTION
**Pull request by shwina**
This PR updates dask-cuda to work with the new `WorkerMemoryManager` abstraction being introduced in  https://github.com/dask/distributed/pull/5904/. Once both PRs are merged, and pending the resolution of https://github.com/dask/distributed/pull/5904/files#r822084806, dask-cuda CI should be unblocked.

### CODE REVIEW

File: dask_cuda/tests/test_proxify_host_file.py [hunk 1]
```diff
@@ -392,18 +392,20 @@ async def test_worker_force_spill_to_disk():
...
                     w = get_worker()
                     # Set a host memory limit that triggers spilling to disk
-                    w.memory_pause_fraction = False
+                    w.memory_manager.memory_pause_fraction = False
                     memory = w.monitor.proc.memory_info().rss
-                    w.memory_limit = memory - 10 ** 8
-                    w.memory_target_fraction = 1
-                    await w.memory_monitor()
+                    w.memory_manager.memory_limit = memory - 10 ** 8
+                    w.memory_manager.memory_target_fraction = 1
```
Comment (crusaderky) [inline dask_cuda/tests/test_proxify_host_file.py:403]: This doesn't work: https://github.com/dask/distributed/issues/5367

**Comment by crusaderky** [comment]
requirements.txt mentions ``distributed>=2021.11.1``; however this PR breaks backwards compatibility with <2022.03.1 (unreleased).

**Comment by jakirkham** [comment]
rerun tests

**Comment by jakirkham** [comment]
rerun tests

**Comment by jakirkham** [comment]
rerun tests

**Comment by jakirkham** [comment]
rerun tests

**Comment by pentschev** [comment]
rerun tests

**Review by pentschev — APPROVED** [review]
All tests passing now, thanks @shwina for the work here!

**Comment by pentschev** [comment]
@gpucibot merge

**Pull request merged**